pdl_syntax.md

```diff
@@ -0,0 +1,828 @@
+}
+```
+
+com/linkedin/pegasus/generator/examples/Fruits.pdl
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * A fruit
+ */
+enum Fruits {
+
+  /**
+   * A red, yellow or green fruit.
+   */
+  APPLE
+
+  /**
+   * A yellow fruit.
+   */
+  BANANA
+
+  /**
+   * An orange fruit.
+   */
+  ORANGE
+
+  /**
+   * A yellow fruit.
+   */
+  PINEAPPLE
+}
+```
+
+com/linkedin/pegasus/generator/examples/MD5.pdl
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * MD5
+ */
+fixed MD5 16
+```
+
+com/linkedin/pegasus/generator/examples/StringList.pdl
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * A list of strings
+ */
+record StringList {
+  element: string
+  next: optional 
+
+    /**
+     * A list of strings
+     */
+    record StringList {
+      element: string
+      next: optional StringList
+    }
+}
+```
+
+com/linkedin/pegasus/generator/examples/InlinedExample.pdl
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * Example on how you can declare an enum and a record inside another record
+ */
+record InlinedExample {
+
+  /**
+   * This is how we inline enum declaration without creating a new pdl file
+   */
+  @symbolDocs.FOO = "It's a foo!"
+  @symbolDocs.NONE = "It's a baz!"
+  @symbolDocs.HASH = "It's a bar!"
+  myEnumField: enum EnumDeclarationInTheSameFile {
+    FOO
+    BAR
+    BAZ
+  }
+
+  /**
+   * A regular string
+   */
+  stringField: string
+
+  /**
+   * A regular int
+   */
+  intField: int
+
+  /**
+   * In this example we will declare a record and an enum inside a union
+   */
+  UnionFieldWithInlineRecordAndEnum: optional union[record myRecord {
+
+    /**
+     * random int field
+     */
+    foo1: int
+
+    /**
+     * random int field
+     */
+    foo2: int
+  }, 
+    /**
+     * Random enum
+     */
+    enum anotherEnum {
+
+      /**
+       * description about FOOFOO
+       */
+      FOOFOO
+
+      /**
+       * description about BARBAR
+       */
+      BARBAR
+    }]
+}
+```
+
+## Record Type
+Records contain any number of fields, which can be primitive type, enums, unions, maps and arrays.
+
+A basic record type can contain a few fields.
+
+For example:
+```
+import org.example.time.DateTime
+
+record Example {
+  field1: string
+  field2: optional int
+  field3: DateTime
+}
+```
+Record fields can be optional.
+
+For example :
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * A foo record
+ */
+record Foo {
+   field1: string
+   field2: optional string
+}
+```
+
+Record fields may have default values.
+
+For example :
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * A foo record
+ */
+record Foo {
+   field1: string
+   field2: string = "message"
+}
+```
+
+A optional field may have default value.
+
+For example : 
+```
+namespace com.linkedin.pegasus.generator.test.idl.records
+
+record WithOptionalPrimitiveDefault {
+  intWithDefault: optional int = 1
+}
+```
+
```

Comment:
Include a section for inlined records, with examples including how to specify default value.

Reply:
Added inline record section and examples.